libyul/AsmParser.h

```diff
@@ -45,13 +47,19 @@ class Parser: public langutil::ParserBase
 		None, ForLoopPre, ForLoopPost, ForLoopBody
 	};
 
+	/// Constructs a Yul parser.
+	///
+	/// @p _charStreamForSourceIndex a callback for retrieving the CharStream related to a source
+	///                              index (see \@src-parsing).
 	explicit Parser(
 		langutil::ErrorReporter& _errorReporter,
 		Dialect const& _dialect,
-		std::optional<langutil::SourceLocation> _locationOverride = {}
+		std::optional<langutil::SourceLocation> _locationOverride = {},
+		std::optional<std::map<unsigned, std::shared_ptr<langutil::CharStream>>> _sourceMap = std::nullopt
```

Comment:
I think `sourceMap` is can be easily confused with the bytecode-to-source-mapping

Also please document that this changes the way the parser works with regards to source locations.